http-gateways/PATH_GATEWAY.md

```diff
@@ -0,0 +1,541 @@
+
+Final array of roots:
+`X-Ipfs-Roots: bafybeiaysi4s6lnjev27ln5icwm6tueaw2vdykrtjkwiphwekaywqhcjze,bafybeihn2f7lhumh4grizksi2fl233cyszqadkn424ptjajfenykpsaiw4,bafkreibn6euazfvoghepcm4efzqx5l3hieof2frhp254hio5y7n3hv5rma`
+
+NOTE: while the first CID will change every time any article is changed,
+the last root (responsible for specific article or a subdirectory) may not
+change at all, allowing for smarter caching beyond what standard Etag offers.
+
+<!-- TODO: https://github.com/ipfs/in-web-browsers/issues/194
+- `X-Ipfs-DagSize`
+    - Indicates the total size of the DAG (raw data + IPLD metadata) representing the requested resource.
+        - For UnixFS this is equivalent  to  `CumulativeSize` from   `ipfs files stat`
+- `X-Ipfs-DataSize`
+    - Indicates the original byte size of the raw data (not impacted by HTTP transfer encoding or compression), without IPFS/IPLD metadata.
+        - For UnixFS this is equivalent to `Size` from `ipfs files stat` or `ipfs dag stat`
+-->
+
+## Response Payload
+
+Data sent with HTTP response depends on the type of requested IPFS resource:
+
+- UnixFS (implicit default)
+  - File
+    - Bytes representing file contents
+  - Directory
+    - Generated HTML with directory index, and/or link to CAR with directory DAG
+    - When `index.html` is present, gateway can skip generating directory index and return it instead
+- Raw block
+  - Opaque bytes, see [application/vnd.ipld.raw](https://www.iana.org/assignments/media-types/application/vnd.ipld.raw)
+- CAR
+  - CAR file or stream, see [application/vnd.ipld.car](https://www.iana.org/assignments/media-types/application/vnd.ipld.car)
+<!-- TODO: https://github.com/ipfs/go-ipfs/issues/8823
+- dag-json / dag-cbor
+  - See [https://github.com/ipfs/go-ipfs/issues/8823](https://github.com/ipfs/go-ipfs/issues/8823)
+-->
+
+# Appendix: notes for implementers
+
+## Content resolution
+
+Content resolution is a process of turning an HTTP request into an IPFS content
+path, and then traversing it until the content identifier (CID) is found.
+
+### Finding the content root
+
+Path Gateway decides what content to serve by taking the path from the URL
+requested and splitting it into two parts: the *CID*  and the *remainder* of
+the path.
+
+The *CID* provides the starting point, often called *content root*. The
+*remainder* of the path, if present,  will be used as instructions to traverse
+IPLD data, starting from that data which the CID identified.
+
+**Note:** Other types of gateway may allow for passing CID by other means, such
+as `Host` header, removing the need for path splitting. (See [
+
+### Traversing remaining path
+
+UnixFS pathing over files and directories is the implicit default used for
+resolving content paths that start with `/ipfs/` and `/ipns/`. It allows for
+traversal based on link names,  which provides a better user experience than
+low level logical pathing from IPLD:
+
```

Comment:
Should we recommend similar behaviour for case where given path is not valid?

Reply:
@vasco-santos mind elaborating?

Reply:
Sorry for not being clear. For instance, if I provide `/ipfs/bafy...foo/this/path/does/not/exist`, this path won't exist. My suggestion here would be if we should recommend a response as part of the spec so that each gateway send same response back.

I would love to have a better response then what we get as https://dweb.link/ipfs/bafkreidyeivj7adnnac6ljvzj2e3rd5xdw3revw4da7mx2ckrstapoupoq/test

Reply:
@vasco-santos added "Handling traversal errors" section in e3637e5 – lmk what additional info should be added in that section